packages/math/src/pool/concentrated/quotes.ts

```diff
@@ -0,0 +1,210 @@
+import { Dec, Int } from "@keplr-wallet/unit";
+
+import { maxSpotPrice, minSpotPrice, smallestDec } from "./const";
+import { TickOverflowError } from "./errors";
+import { addLiquidity, approxSqrt } from "./math";
+import { makeSwapStrategy } from "./swap-strategy";
+import { tickToSqrtPrice } from "./tick";
+import {
+  LiquidityDepth,
+  QuoteInGivenOutParams,
+  QuoteOutGivenInParams,
+} from "./types";
+
+export const ConcentratedLiquidityMath = {
+  calcOutGivenIn,
+  calcInGivenOut,
+  // calcSpotPrice,
+};
+
+interface SwapState {
+  amountRemaining: Dec;
+  amountCalculated: Dec;
+  inittedTickIndex: number;
+  sqrtPrice: Dec;
+  liquidity: Dec;
+  feeGrowthGlobal: Dec;
+}
+
+function calcOutGivenIn({
+  tokenIn,
+  tokenDenom0,
+  poolLiquidity,
+  inittedTicks,
+  curTick,
+  curSqrtPrice,
+  precisionFactorAtPriceOne,
+  swapFee,
+  priceLimit = new Dec(0),
+}: QuoteOutGivenInParams): Int {
+  const isZeroForOne = tokenIn.denom === tokenDenom0;
+  if (isZeroForOne && priceLimit.equals(new Dec(0))) {
+    priceLimit = minSpotPrice;
+  } else if (!isZeroForOne && priceLimit.equals(new Dec(0))) {
+    priceLimit = maxSpotPrice;
+  }
+
+  const sqrtPriceLimit = approxSqrt(priceLimit);
+  const swapStrategy = makeSwapStrategy(isZeroForOne, sqrtPriceLimit, swapFee);
+  const tokenInAmountSpecified = new Dec(tokenIn.amount);
+
+  const curTickWithNetLiq = inittedTicks.find(({ tickIndex }) =>
+    tickIndex.equals(curTick)
+  );
+  if (!curTickWithNetLiq) {
+    throw new Error("curTickNet not found in inittedTicks");
+  }
+  const curTickArrIndex = inittedTicks.indexOf(curTickWithNetLiq);
+  if (curTickArrIndex < 0) {
+    throw new Error("curTickWithNetLiq not found in inittedTicks");
+  }
```

Comment:
I'm suggesting returning cur tick index separately from the array of next ticks with their liquidity net amounts.

Please see this for context: https://github.com/osmosis-labs/osmosis/pull/4714/files#r1149769486

Reply:
I removed use of `curTick`

Does that mean we should still init the index with `initTickValue`? Or should the next tick always be 0 regardless of strategy?